src/main/java/de/rwth/idsg/steve/repository/impl/WebUserRepositoryImpl.java

```diff
@@ -0,0 +1,226 @@
+        if (currentUser == null) {
+            // This would indicate bad coding somewhere
+            throw new AccessDeniedException(
+                "Can't change password as no Authentication object found in context for current user."
+            );
+        }
+
+        String username = currentUser.getName();
+
+        ctx.update(WEB_USER)
+            .set(WEB_USER.PASSWORD, newPassword)
+            .where(WEB_USER.USERNAME.eq(username))
+            .execute();
+
+        Authentication authentication = createNewAuthentication(currentUser, newPassword);
+        SecurityContext context = this.securityContextHolderStrategy.createEmptyContext();
+        context.setAuthentication(authentication);
+        this.securityContextHolderStrategy.setContext(context);
+    }
+
+    @Override
+    public boolean userExists(String username) {
+        return ctx.selectOne()
+            .from(WEB_USER)
+            .where(WEB_USER.USERNAME.eq(username))
+            .fetchOptional()
+            .isPresent();
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        WebUserRecord record = ctx.selectFrom(WEB_USER)
+            .where(WEB_USER.USERNAME.eq(username))
+            .fetchOne();
+
+        if (record == null) {
+            throw new UsernameNotFoundException(username);
+        }
+
+        return User
+            .withUsername(record.getUsername())
+            .password(record.getPassword())
+            .disabled(!record.getEnabled())
+            .authorities(fromJson(record.getAuthorities()))
+            .build();
+    }
+
+    private String[] fromJson(JSON jsonArray) {
+        try {
+            return jacksonObjectMapper.readValue(jsonArray.data(), String[].class);
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private JSON toJson(Collection<? extends GrantedAuthority> authorities) {
+        Collection<String> auths = authorities.stream()
+            .map(GrantedAuthority::getAuthority)
+            .sorted() // keep a stable order of entries
+            .collect(Collectors.toCollection(LinkedHashSet::new)); // prevent duplicates
+
+        try {
+            String str = jacksonObjectMapper.writeValueAsString(auths);
+            return JSON.jsonOrNull(str);
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Lifted from {@link JdbcUserDetailsManager#validateUserDetails(UserDetails)}
+     */
+    private void validateUserDetails(UserDetails user) {
+        Assert.hasText(user.getUsername(), "Username may not be empty or null");
+        validateAuthorities(user.getAuthorities());
+    }
+
+    /**
+     * Lifted from {@link JdbcUserDetailsManager#validateAuthorities(Collection)}
+     */
+    private void validateAuthorities(Collection<? extends GrantedAuthority> authorities) {
```

Comment:
It could be static

Reply:
done in https://github.com/steve-community/steve/pull/1539/commits/960fa23089f37069dd2765428802d4f6e97f6f15